h/static/scripts/directives/annotation.coffee

```diff
@@ -52,6 +53,20 @@ AnnotationController = [
 
     ###*
     # @ngdoc method
+    # @name annotation.AnnotationController#tagsAutoComplete.
+    # @returns {Promise} immediately resolved to {string[]} -
+    # the tags to show in autocomplete.
+    ###
+    this.tagsAutoComplete = (query) ->
+      deferred = $q.defer()
+
+      filteredTags = tagHelpers.filterTags(query)
+      deferred.resolve filteredTags
+
+      return deferred.promise
```

Comment:
This might be more easily written as simple `$.when(tagHelpers.filterTags(query))`.